### DISCUSSION TITLE
'train' and 'val' losses in the json file

### DISCUSSION THREAD

**Discussion by sakerlrf** [open-mmlab/mmaction2#1164] [Q&A] [answered]
### Printing Losses
Hello,
I am trying to understand the guidelines about hooks. In the **Eval_hook**, we see the rule about _greater_ and _less_keys_, and we see the 'loss' in the latter one.

In the main model script, eg. I3D config, I want to state that I need **train** and **val losses** to be **generated**, and _added in the json file_. Eg we can use the following for the evaluation metrics: 
`evaluation = dict(interval=5, metrics=['top_k_accuracy', 'mean_class_accuracy'])`

**My question is**: 
If one needs to have the losses added in the json file, how will this be done?

Thank you in advance.

**Comment by dreamerlin** [comment, marked as answer]
Try `--cfg-options workflow="[(train,1),(val,1)]"`, you can get the loss in json file.

But we still recommend to use `--validate`

**Reply by sakerlrf** [reply]
Much appreciated! 
Thank you.

**Comment by caiocvsilva** [comment]
Hello,

The option to run the code with ` --validate` does not exist anymore? I am trying to get the train accuracy to be printed, as well the validation loss. But most of the answers here and in the documentation seem to relate to an older version of the code.

Any idea how to do it?

Thank you.
